Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -1069,31 +1215,156 @@ public class ProgramBuilder {
+        var body: GetterFunctionGenerator
+
+        init(_ computedVal: Variable, _ body: @escaping GetterFunctionGenerator) {
+            self.computedVal = computedVal
+            self.body = body
+        }
+    }
+
+    public struct SetterProperty {
+        public typealias SetterFunctionGenerator = (Variable) -> ()
+        var name: String
+        var body: SetterFunctionGenerator
+
+        init(_ name: String, _ body: @escaping SetterFunctionGenerator) {
+            self.name = name
+            self.body = body
+        }
+    }
+
+    public struct ComputedSetterProperty {
+        public typealias SetterFunctionGenerator = (Variable) -> ()
+        var computedVal: Variable
+        var body: SetterFunctionGenerator
+
+        init(_ computedVal: Variable, _ body: @escaping SetterFunctionGenerator) {
+            self.computedVal = computedVal
+            self.body = body
+        }
+    }
+
+    public struct ObjectBuilder {
+        public typealias GetterFunctionGenerator = () -> ()
+        public typealias SetterFunctionGenerator = (Variable) -> ()
+        fileprivate var properties: [Any] = []
+
+        public mutating func addProperty(_ name: String, v: Variable) {
+    		properties.append(Property(name, v))
+    	}
+
+        public mutating func addComputedProperty(_ computedVal: Variable, v: Variable) {
+    		properties.append(ComputedProperty(computedVal, v))
+    	}
+
+        public mutating func addSpreadProperty(_ v: Variable) {
+    		properties.append(SpreadProperty(v))
+    	}
+
+        public mutating func addGetter(_ name: String, _ body: @escaping GetterFunctionGenerator) {
+            properties.append(GetterProperty(name, body))
+        }
+
+        public mutating func addComputedGetter(_ computedVal: Variable, _ body: @escaping GetterFunctionGenerator) {
+            properties.append(ComputedGetterProperty(computedVal, body))
+        }
+
+        public mutating func addSetter(_ name: String, _ body: @escaping SetterFunctionGenerator) {
+            properties.append(SetterProperty(name, body))
+        }
+
+        public mutating func addComputedSetter(_ computedVal: Variable, _ body: @escaping SetterFunctionGenerator) {
+            properties.append(ComputedSetterProperty(computedVal, body))
+        }
     }
 
     @discardableResult
-    public func createObject(with initialProperties: [String: Variable], andSpreading spreads: [Variable]) -> Variable {
-        // CreateObjectWithgSpread expects sorted property names
-        var propertyNames = [String](), propertyValues = [Variable]()
-        for (k, v) in initialProperties.sorted(by: { $0.key < $1.key }) {
-            propertyNames.append(k)
-            propertyValues.append(v)
+    public func createObject(body: (inout ObjectBuilder) -> ()) -> Variable {
+        var builder = ObjectBuilder()
+        body(&builder)
+
+        let instr = perform(BeginObjectDefinition(), withInputs: [])
+
+        for prop in builder.properties {
```

Comment:
I'm wondering if we can simplify this: how about having the `addXYZ` methods of the `ObjectBuilder` just append closures into a list:
```swift
public mutating func addProperty(_ name: String, withValue v: Variable) {
    generators.append({ b in b.perform(CreateProperty(propertyName: name), withInputs: [v]) })
}
```
Then just turn this into
```swift
for generator in builder.generators {
    generator(self)
}
```